tests/test_optimize.py

```diff
@@ -0,0 +1,125 @@
+            ]),
+            np.array([
+                [ 7.63891091e-02,  2.15405373e-02, -3.41982696e-03,  2.53188381e-04],
+                [ 2.15405373e-02,  4.70364052e-02,  1.15001079e-02, -2.54481574e-02],
+                [-3.41982696e-03,  1.15001079e-02,  2.23351957e-02, -1.20333088e-03],
+                [ 2.53188381e-04, -2.54481574e-02, -1.20333088e-03,  4.16876485e-02]
+            ]),
+            np.array([
+                [ 8.10451697e-02,  4.38516766e-02,  2.92782757e-02, -2.10502764e-02],
+                [ 4.38516766e-02,  6.93327446e-02,  2.11569124e-02, -1.86073513e-02],
+                [ 2.92782757e-02,  2.11569124e-02,  1.78660782e-02,  9.70497082e-05],
+                [-2.10502764e-02, -1.86073513e-02,  9.70497082e-05,  3.15165163e-02]
+            ]),
+            np.array([
+                [ 6.76527219e-02,  3.73074541e-02,  5.09590306e-02, -1.00836937e-02],
+                [ 3.73074541e-02,  1.16104678e-01,  6.10609709e-03, -2.10680108e-02],
+                [ 5.09590306e-02,  6.10609709e-03,  5.94432673e-02, -2.50081603e-02],
+                [-1.00836937e-02, -2.10680108e-02, -2.50081603e-02,  4.37255531e-02]
+            ]),
+            np.array([
+                [ 9.59149731e-02,  5.56013733e-03,  6.18633837e-02,  1.85653876e-02],
+                [ 5.56013733e-03,  2.22123065e-01, -8.28885992e-03, -3.23027046e-02],
+                [ 6.18633837e-02, -8.28885992e-03,  1.38344768e-01, -8.44515901e-02],
+                [ 1.85653876e-02, -3.23027046e-02, -8.44515901e-02,  1.15266367e-01]
+            ])
+        ]
+
+        self.b = np.array(
+            [ 2.42430396e-09,  9.63863887e-10,  4.57019491e-10,  2.79965227e-10,
+              9.63863887e-10,  1.16254406e-09,  1.35936442e-09, -6.13130105e-10,
+              4.57019491e-10,  1.35936442e-09,  1.47323901e-09, -1.04452618e-09,
+              2.79965227e-10, -6.13130105e-10, -1.04452618e-09,  1.34539435e-09]
+        )
+
+        self.S = np.array(
+            [[ 1.44049041, -0.48037562,  0.0543872 , -0.31003814],
+             [-0.48037562,  2.03658511,  0.16442574,  1.02140979],
+             [ 0.0543872 ,  0.16442574,  2.07414061,  0.47438118],
+             [-0.31003814,  1.02140979,  0.47438118,  2.07438167]]
+        )
+
+        self.x = np.array(
+            [ 2.09826645e-08,  4.88848450e-09,  9.90094995e-09,  3.40230699e-09,
+              4.88848450e-09,  1.52531654e-08,  2.45757724e-08,  1.31921213e-08,
+              9.90094995e-09,  2.45757724e-08,  2.43585121e-08,  3.97418974e-09,
+              3.40230699e-09,  1.31921213e-08,  3.97418974e-09,  2.03443584e-08]
+        )
+
+    def tearDown(self):
+        pass
+
+    def testSolution(self):
+        self.assertTrue(np.allclose(apply(self.A, self.x), self.b))
+
+    def testConjGradReal(self):
+        covar_coeff, _, _ = conj_grad(lambda x: apply(self.A, x), self.b, self.cg_opt)
+
+        self.assertTrue(np.allclose(covar_coeff, self.x))
+
+    def testConjGradComplex(self):
```

Comment:
We need to be testing not only for a complex right-hand side, but for a complex (positive semidefinite) matrix A.

Reply:
Complex matrix `A_comp` was added in new commit  6f43523.